### ISSUE TITLE
Flakiness in StandardFacetInstallationTest

### ISSUE THREAD

**Issue by elharo** [GoogleCloudPlatform/google-cloud-eclipse#1196]
Likely because we're loading a remote schema we shouldn't be loading in the first place:

```
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 7.794 sec <<< FAILURE! - in com.google.cloud.tools.eclipse.appengine.facets.StandardFacetInstallationTest
testStandardFacetInstallation(com.google.cloud.tools.eclipse.appengine.facets.StandardFacetInstallationTest)  Time elapsed: 7.791 sec  <<< FAILURE!
java.lang.AssertionError: Imported projects have errors
/test-dynamic-web-project/war/WEB-INF/web.xml:1: Referenced file contains errors (http://java.sun.com/xml/ns/javaee/javaee_5.xsd).  For more information, right click on the message in the Problems View and select "Show Details..."
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.google.cloud.tools.eclipse.test.util.project.ProjectUtils.failIfBuildErrors(ProjectUtils.java:131)
	at com.google.cloud.tools.eclipse.test.util.project.ProjectUtils.failIfBuildErrors(ProjectUtils.java:125)
	at com.google.cloud.tools.eclipse.test.util.project.ProjectUtils.importProjects(ProjectUtils.java:112)
	at com.google.cloud.tools.eclipse.appengine.facets.StandardFacetInstallationTest.testStandardFacetInstallation(StandardFacetInstallationTest.java:54)
```

**Comment by briandealwis** [comment]
This is after the souped-up `waitForIdle()` added as part of #1300.
```
testStandardFacetInstallation(com.google.cloud.tools.eclipse.appengine.facets.StandardFacetInstallationTest)  Time elapsed: 15.832 sec  <<< ERROR!
java.lang.IllegalArgumentException: Element not found: /test-dynamic-web-project/.settings.
	at org.eclipse.core.internal.watson.ElementTree.elementNotFound(ElementTree.java:256)
	at org.eclipse.core.internal.watson.ElementTree.deleteElement(ElementTree.java:248)
	at org.eclipse.core.internal.resources.Workspace.deleteResource(Workspace.java:1427)
	at org.eclipse.core.internal.resources.Resource.deleteResource(Resource.java:876)
	at org.eclipse.core.internal.localstore.DeleteVisitor.delete(DeleteVisitor.java:68)
	at org.eclipse.core.internal.localstore.DeleteVisitor.visit(DeleteVisitor.java:151)
	at org.eclipse.core.internal.localstore.UnifiedTree.accept(UnifiedTree.java:109)
	at org.eclipse.core.internal.localstore.FileSystemResourceManager.delete(FileSystemResourceManager.java:351)
	at org.eclipse.core.internal.resources.ResourceTree.internalDeleteFolder(ResourceTree.java:358)
	at org.eclipse.core.internal.resources.ResourceTree.internalDeleteProject(ResourceTree.java:393)
	at org.eclipse.core.internal.resources.ResourceTree.standardDeleteProject(ResourceTree.java:850)
	at org.eclipse.core.internal.resources.Resource.unprotectedDelete(Resource.java:1835)
	at org.eclipse.core.internal.resources.Resource.delete(Resource.java:781)
	at org.eclipse.core.internal.resources.Project.delete(Project.java:329)
	at com.google.cloud.tools.eclipse.appengine.facets.StandardFacetInstallationTest.tearDown(StandardFacetInstallationTest.java:55)
```

**Comment by briandealwis** [comment]
> right click on the message in the Problems View and select "Show Details…"

Briefly looked to see if we can pull out this additional data… It's rather encoded, unfortunately, and there are no tools to generate text.  For details on decoding this problem marker, see `org.eclipse.wst.xml.ui.internal.validation.core.errorinfo.ReferencedFileErrorsHandler`.

**Comment by briandealwis** [comment]
As mentioned on #2083, I discovered that `org.eclipse.jst.standard.schemas` already includes these schemas copied in by #2066. No point duplicating information: we should just add `org.eclipse.jst.standard.schemas` as a dependency to `.test.dependencies`.

**Comment by chanseokoh** [comment]
Just saw this on Kokoro. It is dispatching `web-app_2_5.xsd` through HTTP (while the [error log here](https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/2091#issue-239506055) shows that some xsd files are directly from a bundle.) I guess this is not really fixed?

```
testStandardFacetInstallation(com.google.cloud.tools.eclipse.appengine.facets.StandardFacetInstallationTest)  Time elapsed: 6.072 sec  <<< FAILURE!
java.lang.AssertionError: Imported projects have errors
/test-dynamic-web-project/war/WEB-INF/web.xml:1: Referenced file contains errors (http://java.sun.com/xml/ns/javaee/web-app_2_5.xsd).  For more information, right click on the message in the Problems View and select "Show Details..."
  http://java.sun.com/xml/ns/javaee/web-app_2_5.xsd[109:47]: schema_reference.4: Failed to read schema document 'http://java.sun.com/xml/ns/javaee/javaee_5.xsd', because 1) could not find the document; 2) the document could not be read; 3) the root element of the document is not <xsd:schema>.
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.google.cloud.tools.eclipse.test.util.project.ProjectUtils.failIfBuildErrors(ProjectUtils.java:162)
	at com.google.cloud.tools.eclipse.test.util.project.ProjectUtils.failIfBuildErrors(ProjectUtils.java:156)
	at com.google.cloud.tools.eclipse.test.util.project.ProjectUtils.importProjects(ProjectUtils.java:142)
	at com.google.cloud.tools.eclipse.test.util.project.ProjectUtils.importProjects(ProjectUtils.java:86)
	at com.google.cloud.tools.eclipse.appengine.facets.StandardFacetInstallationTest.testStandardFacetInstallation(StandardFacetInstallationTest.java:71)
```

**Comment by briandealwis** [comment]
Hmm.  Looks like a dependency was dropped somehow:
  - javaee_5.xsd is pulled in by the `org.eclipse.jst.standard.schemas` bundle
  - `org.eclipse.jst.standard.schemas` is pulled in by the `org.eclipse.jst.web_core.feature` feature
  - the `org.eclipse.jst.web_core.feature` feature is included by the `org.eclipse.jst.web_ui.feature` feature
  - `com.google.cloud.tools.eclipse.appengine.validation.test`'s pom.xml pulls in the `org.eclipse.jst.web_ui.feature` feature

Oh no: both `org.eclipse.wst.standard.schemas` nor `org.eclipse.jst.standard.schemas` are not marked as lazily activated, so their `plugin.xml` contributions are ignored until they're explicitly started.

**Comment by briandealwis** [comment]
> Oh no: neither org.eclipse.wst.standard.schemas nor org.eclipse.jst.standard.schemas are marked as lazily activated, so their plugin.xml contributions are ignored until they're explicitly started.
 
Hmm, I was sure this was the case, but apparently not.

**Comment by briandealwis** [comment]
Oops: in the fix for #2190, I appear to have wiped out the schema-bundle-inclusion changes from #2092.
